File: app/src/main/java/tech/pixelw/castrender/CastRenderApp.java
package tech.pixelw.castrender;

import android.content.Context;

import androidx.multidex.MultiDexApplication;

import tech.pixelw.castrender.utils.report.BuglyReporter;
import tech.pixelw.castrender.utils.report.Reporter;

/**
 * @author Carl Su "Pixelw"
 * @date 2021/10/25
 */
public class CastRenderApp extends MultiDexApplication {
    private static Context appContext;

    @Override
    public void onCreate() {
        super.onCreate();
        appContext = getApplicationContext();
        Reporter.INSTANCE.setSolution(new BuglyReporter());
    }

    public static Context getAppContext() {
        return appContext;
    }
}

File: app/src/main/java/tech/pixelw/castrender/CastRenderApp.kt
package tech.pixelw.castrender

import android.content.Context
import androidx.multidex.MultiDexApplication
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import kotlinx.coroutines.cancel
import tech.pixelw.castrender.utils.report.BuglyReporter
import tech.pixelw.castrender.utils.report.Reporter.solution

/**
 * @author Carl Su "Pixelw"
 * @date 2021/10/25
 */
class CastRenderApp : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        appContext = applicationContext
        solution = BuglyReporter()
    }

    companion object {
        @JvmStatic
        lateinit var appContext: Context

        val applicationScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
    }

    override fun onLowMemory() {
        super.onLowMemory()
        applicationScope.cancel("onLowMemory() called")
    }
}